app/v1/controllers/validators/AmendOtherReliefsValidator.scala

```diff
@@ -0,0 +1,202 @@
+    else Invalid(List(error.withPath(path)))
+
+  private def validateFieldLength(field: String, path: String, error: MtdError): Validated[Seq[MtdError], Unit] =
+    if (field.nonEmpty && field.length <= 105) valid
+    else Invalid(List(error.withPath(path)))
+
+  def validateBusinessRules(parsed: AmendOtherReliefsRequestData): Validated[Seq[MtdError], AmendOtherReliefsRequestData] = {
+    import parsed.body._
+
+    val validatedNonDeductibleLoanInterest = nonDeductibleLoanInterest.map(validateNonDeductibleLoanInterest).getOrElse(valid)
+    val validatedPayrollGiving             = payrollGiving.map(validatePayrollGiving).getOrElse(valid)
+    val validatedQualifyingDistributionRedemptionOfSharesAndSecurities =
+      qualifyingDistributionRedemptionOfSharesAndSecurities.map(validateQualifyingDistributionRedemptionOfSharesAndSecurities).getOrElse(valid)
+
+    def zipAndValidate[A](fields: Option[Seq[A]], validate: (A, Int) => Validated[Seq[MtdError], Unit]) =
+      fields
+        .map(
+          _.zipWithIndex
+            .map { case (entry, index) => validate(entry, index) }
+            .sequence
+            .andThen(_ => valid))
+        .getOrElse(valid)
+
+    val validatedPostCessationTradeReliefAndCertainOtherLosses =
+      zipAndValidate(postCessationTradeReliefAndCertainOtherLosses, validatePostCessationTradeReliefAndCertainOtherLosses)
+
+    val validatedMaintenancePayments =
+      zipAndValidate(maintenancePayments, validateMaintenancePayments)
+
+    val validatedAnnualPaymentsMade = annualPaymentsMade.map(validateAnnualPaymentsMade).getOrElse(valid)
+    val validatedQualifyingLoanInterestPayments =
+      zipAndValidate(qualifyingLoanInterestPayments, validateQualifyingLoanInterestPayments)
+
+    combine(
+      validatedNonDeductibleLoanInterest,
+      validatedPayrollGiving,
+      validatedQualifyingDistributionRedemptionOfSharesAndSecurities,
+      validatedPostCessationTradeReliefAndCertainOtherLosses,
+      validatedMaintenancePayments,
+      validatedAnnualPaymentsMade,
+      validatedQualifyingLoanInterestPayments
+    ).onSuccess(parsed)
+  }
+
+  private def validateNonDeductibleLoanInterest(nonDeductibleLoanInterest: NonDeductibleLoanInterest): Validated[Seq[MtdError], Unit] = {
+    import nonDeductibleLoanInterest._
+
+    val validatedCustomerReference =
+      customerReference.map(ref => validateCustomerRef(ref, "/nonDeductibleLoanInterest/customerReference")).getOrElse(valid)
+
+    val validatedReliefClaimed = resolveParsedNumber(reliefClaimed, None, Some("/nonDeductibleLoanInterest/reliefClaimed"))
+
+    combine(validatedCustomerReference, validatedReliefClaimed).andThen(_ => valid)
+  }
+
+  private def validatePayrollGiving(payrollGiving: PayrollGiving): Validated[Seq[MtdError], Unit] = {
+    import payrollGiving._
+
+    val validatedCustomerReference =
+      customerReference.map(ref => validateCustomerRef(ref, "/payrollGiving/customerReference")).getOrElse(valid)
+
```

Comment:
We can actually simplify all these mappings

```suggestion
    val validatedCustomerReference =
      customerReference.map(validateCustomerRef(_, "/payrollGiving/customerReference")).getOrElse(valid)

```

Reply:
(Same for all the map calls, using the underscore notation instead of having to create a named val)